pkg/reconciler/v1alpha1/revision/revision.go

```diff
@@ -263,19 +263,13 @@ func (c *Reconciler) Reconcile(ctx context.Context, key string) error {
 }
 
 func (c *Reconciler) reconcileBuild(ctx context.Context, rev *v1alpha1.Revision) error {
-	if rev.Spec.BuildName == "" {
+	if rev.Spec.BuildRef == nil {
 		return nil
 	}
 
 	logger := commonlogging.FromContext(ctx)
 
-	//TODO(imikushin): get this from rev when it's added to Revision in the API
-	buildRef := corev1.ObjectReference{
-		APIVersion: "build.knative.dev/v1alpha1",
-		Kind:       "Build",
-		Namespace:  rev.Namespace,
-		Name:       rev.Spec.BuildName,
-	}
+	buildRef := *rev.Spec.BuildRef
```

Comment:
Perhaps fall back on populating this from `BuildName` if it's specified and this is not.